src/main/java/com/cloudbees/plugins/credentials/CredentialsStore.java

```diff
@@ -197,12 +217,12 @@ public boolean hasPermission(@NonNull Authentication a, @NonNull Permission perm
      * Note: This is just a convenience function.
      * </p>
      *
-     * @throws org.acegisecurity.AccessDeniedException if the user doesn't have the permission.
+     * @throws AccessDeniedException3 if the user doesn't have the permission.
      */
     public final void checkPermission(@NonNull Permission p) {
-        Authentication a = Jenkins.getAuthentication();
-        if (!hasPermission(a, p)) {
-            throw new AccessDeniedException2(a, p);
+        Authentication a = Jenkins.getAuthentication2();
+        if (!hasPermission2(a, p)) {
+            throw new AccessDeniedException3(a, p);
```

Comment:
Note that this will cause a change of behavior even without changing any clients. Not much we can do about that; one of the tricky aspects of JEP-222 was providing compatibility for exception types. I updated at least the standard handlers to accept either the Acegi or Spring Security versions, and automatically proxied from some other methods.

Reply:
`checkPermission2(..)`?